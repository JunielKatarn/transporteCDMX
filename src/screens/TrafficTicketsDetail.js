import React, { Fragment } from 'react'
import { FlatList, Text, View } from 'react-native'
import { Caption } from 'react-native-paper'
import TrafficTicket from '../components/TrafficTicket'
import { keyExtractor } from '../utils/list'
import { sharedStyles } from '../styles'
import { translate } from '../i18n'

const _renderItem = ({ item }) => <TrafficTicket ticket={item} />
// TODO: We need a nice illustration for this
const _renderEmptyPlaceholder = () => (
  <View>
    <Text>Empty</Text>
  </View>
)

const _getCountText = ticketsCount => {
  if (ticketsCount === 1) {
    return translate('one_traffic_ticket')
  }
  return translate('multiple_traffic_tickets').replace('_', ticketsCount)
}

const TrafficTicketsDetail = props => {
  const { trafficTickets } = props.route.params

  // TODO: We need to create a dictionary of year -> UMA relationship
  const _renderTotalInfractions = () => {
    if (trafficTickets?.length) {
      return (
        <Fragment>
          <View style={[sharedStyles.alignItemsCenter, sharedStyles.py2]}>
            <Caption>{_getCountText(trafficTickets.length)}</Caption>
            <Caption>{translate('fine_calculation')}</Caption>
          </View>
        </Fragment>
      )
    }
    return null
  }

  return (
    <FlatList
      data={trafficTickets}
      renderItem={_renderItem}
      keyExtractor={keyExtractor}
      ListHeaderComponent={_renderTotalInfractions}
      ListEmptyComponent={_renderEmptyPlaceholder}
    />
  )
}

export default TrafficTicketsDetail
