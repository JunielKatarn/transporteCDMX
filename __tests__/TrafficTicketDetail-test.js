import React from 'react'
import { shallow } from 'enzyme'
import TrafficTicketsDetail from '../src/screens/TrafficTicketsDetail'
import trafficTicket from './fixtures/trafficTicket-fixture'

const params = { params: { trafficTickets: [trafficTicket] } }
const emptyParams = { params: { trafficTickets: [] } }

describe('TrafficTicketsDetail', () => {
  it('it renders correctly WITH Traffic Tickets', () => {
    const wrapper = shallow(<TrafficTicketsDetail route={params} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('it renders correctly WITHOUT Traffic Tickets', () => {
    const wrapper = shallow(<TrafficTicketsDetail route={emptyParams} />)
    expect(wrapper).toMatchSnapshot()
  })
})
