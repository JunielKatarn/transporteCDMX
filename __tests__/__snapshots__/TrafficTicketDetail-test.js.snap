// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrafficTicketsDetail it renders correctly WITH Traffic Tickets 1`] = `
<FlatList
  ListEmptyComponent={[Function]}
  ListHeaderComponent={[Function]}
  data={
    Array [
      Object {
        "causeDetail": "This a cause",
        "fine": 2020,
        "folio": 1234567,
        "formattedDate": "2020-01-01",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`TrafficTicketsDetail it renders correctly WITHOUT Traffic Tickets 1`] = `
<FlatList
  ListEmptyComponent={[Function]}
  ListHeaderComponent={[Function]}
  data={Array []}
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
